src/parsimmon.js

```diff
@@ -396,14 +396,16 @@ _.then = function(next) {
 
 _.many = function() {
   var self = this;
-
   return Parsimmon(function(input, i) {
     var accum = [];
     var result = undefined;
-
+    
     for (;;) {
       result = mergeReplies(self._(input, i), result);
       if (result.status) {
+        if (i === result.index) {
+          return makeFailure(i, 'Infinite loop has been detected in many() parser');
+        }
```

Comment:
this is off topic but i wish if parsimmon had custom errors containing its own kind...